### PR TITLE
Add checklist to pull request template

### PR DESCRIPTION
**Pull request by mehertz**
Renders as...

image

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -15,10 +15,23 @@ is merged.
...
 
+#### Checklist
+
+<details>
+  <summary>
+   Code change checklist...
+  </summary>
+ 
+ - [ ] Have you updated the relevant docstring/documentation pages?
+ - [ ] Does your change support the following schema options:
```
[the remark this answers is not in the document]
Comment (mehertz) [reply .github/PULL_REQUEST_TEMPLATE.md]: Good idea!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -15,10 +15,24 @@ is merged.
...
+  <summary>
+   Code change checklist...
+  </summary>
+ 
+ - [ ] Have you updated the relevant docstring/documentation pages?
+ - [ ] Does your change have automated test coverage for the following schema options:
+   - [ ] Dynamic schema
+   - [ ] Sparse data
+ - [ ] Do all exceptions introduced raise appropriate [error messages](https://docs.arcticdb.io/error_messages/)?
+ - [ ] Are API changes highlighted in the above section?
```
Comment (jjerphan) [inline .github/PULL_REQUEST_TEMPLATE.md:33]:
```suggestion
 - [ ] Are API changes highlighted in the above section?
 - [ ] Is this contribution tested against [all ArcticDB's features](docs/mkdocs/docs/technical/contributing.md)?
```

Comment (mehertz) [reply .github/PULL_REQUEST_TEMPLATE.md:33]: I could remove the dynamic schema/sparse data bit with that?

My problem is that's a long page. And realistically people won't pay attention to every line on that to check a box - I think it's too detailed to be of value in a PR checklist in other words. 

Equally my list is incomplete, so is of limited value in that direction. 

I'm not sure what's most useful in practise.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -15,10 +15,24 @@ is merged.
...
 
+#### Checklist
+
+<details>
+  <summary>
+   Code change checklist...
+  </summary>
+ 
+ - [ ] Have you updated the relevant docstring/documentation pages?
+ - [ ] Does your change have automated test coverage for the following schema options:
```
Comment (poodlewars) [inline .github/PULL_REQUEST_TEMPLATE.md:29]: Refer to AlexO's docs page instead of this, I think

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -15,10 +15,24 @@ is merged.
...
 
 
+#### Checklist
+
+<details>
+  <summary>
+   Code change checklist...
+  </summary>
+ 
+ - [ ] Have you updated the relevant docstring/documentation pages?
```
Comment (poodlewars) [inline .github/PULL_REQUEST_TEMPLATE.md:28]:
```suggestion
 - [ ] Have you updated the relevant docstrings and documentation?
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (poodlewars) [reply .github/PULL_REQUEST_TEMPLATE.md:33]: Prefer Julien's suggestion

**Review by poodlewars — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (poodlewars) [inline .github/PULL_REQUEST_TEMPLATE.md:33]:
```suggestion
 - [ ] Are API changes highlighted in the PR description?
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -15,10 +15,24 @@ is merged.
...
 #### What does this implement/fix? Explain your changes.
 
 
+#### Checklist
+
+<details>
+  <summary>
+   Code change checklist...
+  </summary>
+
```
Comment (poodlewars) [inline .github/PULL_REQUEST_TEMPLATE.md:27]: Is the PR labelled correctly as feature or bug so it appears in the autogenerated release notes?

**Review by poodlewars — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (jjerphan) [reply .github/PULL_REQUEST_TEMPLATE.md:33]: I think that if the review process is properly done, we should not be too concerned with the last item that I proposed. What do you think?

**Review by mehertz — COMMENTED** [review]
Address review comments

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (mehertz) [reply .github/PULL_REQUEST_TEMPLATE.md]: I've changed to be a link the @alexowens90 page.

**Review by jjerphan — APPROVED** [review]
Thank you, @mehertz.

**Pull request merged**